ablySpec/RealtimeClientChannel.swift

```diff
@@ -13,6 +13,32 @@ class RealtimeClientChannel: QuickSpec {
     override func spec() {
         describe("Channel") {
 
+            // RTL4
+            describe("attach") {
+
+                // RTL4a
+                it("if already ATTACHED or ATTACHING nothing is done") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    var errorInfo: ARTErrorInfo?
+                    let channel = client.channel("test")
+
+                    errorInfo = channel.attach()
+                    expect(errorInfo).to(beNil())
+                    expect(channel.state).to(equal(ARTRealtimeChannelState.Attaching))
+
+                    errorInfo = channel.attach()
+                    expect(errorInfo!.message).to(contain("Already attached"))
```

Comment:
I think this goes beyond the API change. This should _not_ return an error, even with this API.